pco_python/src/lib.rs

```diff
@@ -0,0 +1,76 @@
+use numpy::{Element, PyArrayDyn};
+use pyo3::exceptions::PyRuntimeError;
+use pyo3::prelude::{pymodule, FromPyObject, PyModule, PyObject, PyResult, Python};
+use pyo3::types::PyBytes;
+
+use pco::data_types::NumberLike;
+use pco::standalone::{auto_compress, simple_decompress_into};
+use pco::DEFAULT_COMPRESSION_LEVEL;
+
+// The Numpy crate recommends using this type of enum to write functions that accept different Numpy dtypes
+// https://github.com/PyO3/rust-numpy/blob/32740b33ec55ef0b7ebec726288665837722841d/examples/simple/src/lib.rs#L113
+#[derive(FromPyObject)]
+enum ArrayDynFloat<'py> {
+  F32(&'py PyArrayDyn<f32>),
+  F64(&'py PyArrayDyn<f64>),
+  I32(&'py PyArrayDyn<i32>),
+  I64(&'py PyArrayDyn<i64>),
+  U32(&'py PyArrayDyn<u32>),
+  U64(&'py PyArrayDyn<u64>),
+}
+
+fn compress_typed<T: NumberLike + Element>(py: Python, arr: &PyArrayDyn<T>) -> PyResult<PyObject> {
+  let arr_ro = arr.readonly();
+  let src = arr_ro.as_slice()?;
+  let compressed = auto_compress(src, DEFAULT_COMPRESSION_LEVEL);
+  Ok(PyBytes::new(py, &compressed).into())
+}
+
+fn decompress_typed<T: NumberLike + Element>(
+  compressed: &PyBytes,
+  out: &PyArrayDyn<T>,
+) -> PyResult<()> {
+  let mut out_rw = out.readwrite();
+  let dst = out_rw.as_slice_mut()?;
+  let src = compressed.as_bytes();
+  let progress = simple_decompress_into(src, dst)
+    .map_err(|e| PyRuntimeError::new_err(format!("pco decompression error: {}", e)))?;
+  if !progress.finished {
+    Err(PyRuntimeError::new_err(
+      "decompression didn't finish. Buffer too small?",
+    ))
+  } else {
+    Ok(())
+  }
+}
+
+#[pymodule]
+fn pcodec(_py: Python<'_>, m: &PyModule) -> PyResult<()> {
+  m.add("__version__", env!("CARGO_PKG_VERSION"))?;
+
+  #[pyfn(m)]
+  fn compress<'py>(py: Python<'py>, x: ArrayDynFloat<'py>) -> PyResult<PyObject> {
```

Comment:
Let's name these after the Rust functions they wrap - `auto_compress` and `simple_decompress_info`.